AdminServer/appscale/admin/__init__.py

```diff
@@ -507,10 +507,16 @@ def delete(self, project_id, service_id):
 class VersionsHandler(BaseHandler):
   """ Manages service versions. """
 
-  # A rule for validating version IDs.
-  VERSION_ID_RE = re.compile(r'(?!-)[a-z\d\-]{1,100}')
+  # A rule for validating project IDs.
   PROJECT_ID_RE = re.compile(r'^[a-z][a-z\d\-]{5,29}$')
 
+  # A rule for validating version IDs.
+  VERSION_ID_RE = re.compile(r'^(?:^(?!-)[a-z\d\-]{0,62}[a-z\d]$)$')
```

Comment:
I think replacing \d with 0-9 makes the regular expression a little more readable. Also you shouldn't have to include the re in (?:) Also when '-' is used at the end of the set of characters you don't need to escape it
```suggestion
  VERSION_ID_RE = re.compile(r'^(?!-)[a-z0-9-]{0,62}[a-z0-9]$')
```

Reply:
Thank you, Steven! This re was suggested by gcloud, but I also find it a little bit tricky :)